src/pydicom/dataelem.py

```diff
@@ -429,7 +457,12 @@ def json_dump(d: dict[str, Any]) -> str:
 
     @property
     def value(self) -> Any:
-        """Return the element's value."""
+        """Return the element's value. If the value is a Buffer, the buffer will be read and returned.
+        To read the buffer in chunks, see value_generator."""
+        if self.is_buffered:
+            self._value = cast(BufferedIOBase, self._value).read()
+            self.is_buffered = False
+
         return self._value
 
```

Comment:
I think we need to update the `value` setter code as well - `value` could be set to a buffer after initial creation. The logic to check that on __init__ probably belows in the `value` setter, actually, where it can handle both cases.